core/embed/rust/src/ui/debug.rs

```diff
@@ -0,0 +1,219 @@
+//! Including some useful debugging features,
+//! like printing of the struct details.
+
+use heapless::String;
+
+use super::{
+    component::{
+        pad::Pad,
+        text::{
+            common::TextBox,
+            layout::{Span, TextLayout},
+        },
+    },
+    display::{Color, Font, Icon},
+    geometry::{Grid, Insets, Offset, Point, Rect},
+};
+use crate::{micropython::buffer::StrBuffer, time::Duration};
+
+// NOTE: not defining a common trait, like
+// Debug {fn print(&self);}, so that the trait does
+// not need to be imported when using the
+// print() function. It suits the use-case of being quickly
+// able to use the print() for debugging and then delete it.
```

Comment:
I'm not sure the ease of use justifies having to have every member of supported structs as `pub`. Let's keep it if you find it useful, but personally I'd prefer to avoid maintaining hand-written serialization functions. Wouldn't writing a `Tracer` impl that produces single line strings work just as well?

Reply:
worth considering bringing in ufmt for the `uDebug` trait

Reply:
I made these debug things a long time ago, and almost do not use them nowadays, so we can maybe freely delete them. 

> Wouldn't writing a Tracer impl that produces single line strings work just as well?

Will think about this

> worth considering bringing in ufmt for the uDebug trait

I don't know how I would do this